modules/appeals_api/spec/services/appeals_api/pdf_construction/generator_spec.rb

```diff
@@ -20,96 +20,100 @@
-            extra_nod_v2.form_data = data
+        context 'with extra content' do
+          let(:fixture_name) { 'expected_10182_extra.pdf' }
+          let(:nod) { create(:extra_notice_of_disagreement_v2, created_at: '2021-02-03T14:15:16Z') }
 
-            generated_pdf = described_class.new(extra_nod_v2, pdf_version: 'v2').generate
-            expected_pdf = fixture_filepath('expected_10182_extra.pdf', version: 'v2')
+          it 'generates the expected pdf' do
             expect(generated_pdf).to match_pdf expected_pdf
-            File.delete(generated_pdf) if File.exist?(generated_pdf)
           end
         end
 
-        context 'pdf minimal content verification' do
-          let(:minimal_nod_v2) { create(:minimal_notice_of_disagreement_v2, created_at: '2021-02-03T14:15:16Z') }
+        context 'with minimal content' do
+          let(:fixture_name) { 'expected_10182_minimal.pdf' }
+          let(:nod) { create(:minimal_notice_of_disagreement_v2, created_at: '2021-02-03T14:15:16Z') }
 
           it 'generates the expected pdf' do
-            generated_pdf = described_class.new(minimal_nod_v2, pdf_version: 'v2').generate
-            expected_pdf = fixture_filepath('expected_10182_minimal.pdf', version: 'v2')
             expect(generated_pdf).to match_pdf expected_pdf
-            File.delete(generated_pdf) if File.exist?(generated_pdf)
           end
         end
 
         context 'pdf max length content verification' do
-          let(:schema) { read_schema('10182.json', 'v2') }
-          let(:nod) { build(:extra_notice_of_disagreement_v2, created_at: '2021-02-03T14:15:16Z') }
-          let(:data) { override_max_lengths(nod, schema) }
+          let(:fixture_name) { 'expected_10182_maxlength.pdf' }
+          let(:nod) do
+            build(:extra_notice_of_disagreement_v2, created_at: '2021-02-03T14:15:16Z') do |appeal|
+              appeal.form_data = override_max_lengths(appeal, read_schema('10182.json', 'v2'))
+              appeal.auth_headers.merge!(
+                {
+                  'X-VA-SSN' => 'W' * 9,
+                  'X-VA-First-Name' => 'W' * 30,
+                  'X-VA-Middle-Initial' => 'W',
+                  'X-VA-Last-Name' => 'W' * 40,
+                  'X-VA-NonVeteranClaimant-First-Name' => 'W' * 30,
+                  'X-VA-NonVeteranClaimant-Middle-Initial' => 'W',
+                  'X-VA-NonVeteranClaimant-Last-Name' => 'W' * 40,
+                  'X-VA-File-Number' => 'W' * 9,
+                  'X-Consumer-Username' => 'W' * 255,
+                  'X-Consumer-ID' => 'W' * 255
+                }
+              )
+            end
+          end
 
           # TODO: Try to figure out why the CI runner interprets our expected pdf differently than locally, despite
           #       being visually identical.
           # e.g. on CI, some text is interpreted in a slightly different order or W's are added in odd places.
           xit 'generates the expected pdf' do
```

Comment:
I did un-skip this example and run it for pdfs v2 and v3, looking at the generated PDFS, too - things were looking normal locally, and it still fails if it runs in github actions.